File: Java/2.两数之和.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        if(l1 == null) return l2;
        if(l2 == null) return l1;
        ListNode res = new ListNode((l1.val + l2.val)%10);
        res.next = addTwoNumbers(l1.next,l2.next);
        if((l1.val + l2.val)/10 == 1){
            res.next = addTwoNumbers(res.next,new ListNode(1));
        }
        return res;
    }
}

File: Python/2.两数之和.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        if l1 is None:
            return l2
        if l2 is None:
            return l1
        res = ListNode((l1.val + l2.val)%10)
        res.next = self.addTwoNumbers(l1.next,l2.next)
        if (l1.val + l2.val)/10 == 1:
            res.next = self.addTwoNumbers(res.next,ListNode(1))
        return res